building_sim_plugins/building_gazebo_plugins/src/toggle_floors.cpp

```diff
@@ -45,6 +45,17 @@ class ToggleFloors : public gazebo::GUIPlugin
       string floor_name = floor_ele->GetAttribute("name")->GetAsString();
       string model_name =
         floor_ele->GetAttribute("model_name")->GetAsString();
+
+      std::vector<string> models;
+      for (sdf::ElementPtr model_ele = floor_ele->GetFirstElement();
+        model_ele;
+        model_ele = model_ele->GetNextElement("model"))
+      {
+        if (model_ele->GetName() != string("model"))
```

Comment:
I think we can get away with checking for this if we change the structure of the loop.
```
auto model_ele = floor_ele->GetElement("model")
while (model_ele)
{
  if (model_ele.HasAttribute("name")
    models.push_back(model_ele->GetAttribute("name")->GetAsString());
  model_ele = model_ele->GetNextElement("model");
}
```